superset/models/helpers.py

```diff
@@ -116,12 +116,48 @@ def __init__(  # noqa
         self.status = status
         self.error_message = error_message
 
+def merge_perm(sm, permission_name, view_menu_name, connection):
```

Comment:
Yes functionality-wise. But it doesn't seem to work with listener. I got an error saying "use connection level operation". So the merge function here uses connection.